exist-core/src/main/java/org/exist/xquery/value/DoubleValue.java

```diff
@@ -506,21 +460,17 @@ public <T> T toJavaObject(final Class<T> target) throws XPathException {
             return (T) Boolean.valueOf(effectiveBooleanValue());
         }
 
-        throw new XPathException(getExpression(), 
-                "cannot convert value of type "
-                        + Type.getTypeName(getType())
-                        + " to Java object of type "
-                        + target.getName());
+        throw new XPathException(getExpression(), ErrorCodes.FORG0001, "cannot convert value of type "
```

Comment:
Is `FORG0001` the correct error code here?

Reply:
It's the code throughout the class for the same type of error. So I assume that is the case.